kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js

```diff
@@ -494,10 +495,34 @@ define([
 
                 ui.setContent('input-container', content);
                 const dropdown = $(ui.getElement('input-container.input'));
+                const data = [];
+                if (config.isViewOnly) {
```

Comment:
This block here sets up initial data, it's really the main difference in view-only mode (i.e. `dynamicDropdownView.js`. 

Select2 does a fun thing where you can give it a `data` parameter, and it'll use that, formatted as helpful as you like. Mixing that with `ajax` calls makes for some trickiness. BUT if we're in the view version, it doesn't matter much.